hash.py

```diff
@@ -0,0 +1,23 @@
+class Hash(object):
+    def __init__(self, buckets):
+        self.heap = []
```

Comment:
be careful about using this term, as  `heap` has a *very* specific meaning in computer science.  It's misleading to call this a heap, as it is really just a list.